returnn/tf/layers/base.py

```diff
@@ -491,18 +492,26 @@ def transform_config_dict(cls, d, network, get_layer):
           if target.startswith("layer:"):
             target_layers[target] = get_layer(target[len("layer:"):])
           else:
-            # Note: This is a workaround for cases where we need to know about used data keys before the layer
-            # itself is constructed (e.g. in _SubnetworkRecCell.get_output).
-            # A nicer solution would be to not modify this here,
-            # but instead lazily handle it in TFNetwork.get_extern_data,
-            # such that we do not need to know in advance which data keys we need.
-            # Also, if we are inside a rec layer, and doing search, we also cannot do that.
-            if network.is_inside_rec_layer() and not network.search_flag:
-              network.get_extern_data(target, mark_data_key_as_used=True)
-            if not network.search_flag:
-              # Also, there are cases when we want to have the target as an explicit layer dep,
-              # e.g. when the target has a beam, to derive the search choices.
-              target_layers[target] = get_layer("data:%s" % target)
+            try:
+              # Note: This is a workaround for cases where we need to know about used data keys before the layer
+              # itself is constructed (e.g. in _SubnetworkRecCell.get_output).
+              # A nicer solution would be to not modify this here,
+              # but instead lazily handle it in TFNetwork.get_extern_data,
+              # such that we do not need to know in advance which data keys we need.
+              # Also, if we are inside a rec layer, and doing search, we also cannot do that.
+              if network.is_inside_rec_layer() and not network.search_flag:
+                network.get_extern_data(target, mark_data_key_as_used=True)
+              else:
+                # Anyway, to check for the name.
+                network.get_extern_data(target, mark_data_key_as_used=False)
+            except DataNotFound:
+              # Expect that this is a layer.
+              target_layers[target] = get_layer(target)
+            else:
```

Comment:
where does this `else` belong to?

Reply:
Ah nevermind, I did not know that there is try-except-else and not only try-except-finally